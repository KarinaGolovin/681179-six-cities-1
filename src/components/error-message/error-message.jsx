import React from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';

export const ErrorMessage = ({message = ``}) => {
  return <div className={`error-message ${message ? `error-message--active` : ``}`}>
    <div className={`error-message__content`}>
      <p className="error-message__text">{message}</p>
    </div>
  </div>;
};

ErrorMessage.propTypes = {
  message: PropTypes.string
};

const mapStateToProps = (state) => {
  return {
    message: state.notifications ? state.notifications.error.message : null
  };
};

export default connect(mapStateToProps)(ErrorMessage);
