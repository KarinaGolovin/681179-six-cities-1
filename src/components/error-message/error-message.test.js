import React from 'react';
import renderer from 'react-test-renderer';
import {ErrorMessage} from '../error-message/error-message.jsx';

it(`ErrorMessage correctly renders`, () => {
  const component = renderer.create(
      <ErrorMessage message={`Oooops`}/>
  ).toJSON();

  expect(component).toMatchSnapshot();
});
