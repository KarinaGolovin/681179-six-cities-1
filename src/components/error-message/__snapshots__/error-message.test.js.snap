// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage correctly renders 1`] = `
<div
  className="error-message error-message--active"
>
  <div
    className="error-message__content"
  >
    <p
      className="error-message__text"
    >
      Oooops
    </p>
  </div>
</div>
`;
